internal/rbd/rbd_util.go

```diff
@@ -1165,13 +1225,42 @@ func (rv *rbdVolume) resize(newSize int64) error {
 	}
 	defer image.Close()
 
+	thick, err := rv.isThickProvisioned()
+	if err != nil {
+		return err
+	}
+
+	// offset is used to track from where on the expansion is done, so that
+	// the extents can be allocated in case the image is thick-provisioned
+	var offset uint64
+	if thick {
+		st, statErr := image.Stat()
+		if statErr != nil {
+			return statErr
+		}
+
+		offset = st.Size
+	}
+
 	err = image.Resize(uint64(util.RoundOffVolSize(newSize) * helpers.MiB))
 	if err != nil {
 		return err
 	}
 
+	if thick {
+		err = rv.allocate(offset)
+		if err != nil {
+			resizeErr := image.Resize(uint64(util.RoundOffVolSize(newSize) * helpers.MiB))
```

Comment:
@nixpanic the resize as to be done for old size right if allocation fails?

Reply:
Oh, yes, `newSize` is not correct, that should be the old size!